Backend/WebApi.Weather/WeatherClient.cs

```diff
@@ -0,0 +1,142 @@
+﻿
+using System.Globalization;
+using System.Text.Json;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using static Microsoft.AspNetCore.Http.QueryString;
+
+namespace WebApi.Weather;
+
+public interface IWeatherClient
+{
+    Task<WeatherForecastApi> GetWeatherForecastAsync(WeatherForecastOptionsApi optionsApi);
+}
+
+public class WeatherClient(ILogger<WeatherClient> logger, HttpClient client) : IWeatherClient
+{
+    
+    private readonly string _weatherApiUrl = "https://api.open-meteo.com/v1/forecast";
+
+
+
+    public async Task<WeatherForecastApi?> QueryAsync(WeatherForecastOptionsApi optionsApi)
+    {
+        try
+        {
+            return await GetWeatherForecastAsync(optionsApi);
+        }
+        catch (Exception)
+        {
+            logger.LogError("Not Working");
+            return null;
+        }
+    }
+
+    /// <summary>
+    ///     Performs one GET-Request to get weather information
+    /// </summary>
+    /// <param name="latitude">City latitude</param>
+    /// <param name="longitude">City longitude</param>
+    /// <returns>Awaitable Task containing WeatherForecast or NULL</returns>
+    public async Task<WeatherForecastApi?> QueryAsync(float latitude, float longitude)
+    {
+        var options = new WeatherForecastOptionsApi
+        {
+            Latitude = latitude,
+            Longitude = longitude
+        };
+        return await QueryAsync(options);
+    }
+
+    public WeatherForecastApi? Query(WeatherForecastOptionsApi optionsApi)
+    {
+
+        return QueryAsync(optionsApi).GetAwaiter().GetResult();
+    }
+
+    public WeatherForecastApi? Query(float latitude, float longitude)
+    {
+        return QueryAsync(latitude, longitude).GetAwaiter().GetResult();
+    }
+
+
+    public string WeatherCodeToString(int weatherCode)
+    {
+        if (Enum.IsDefined(typeof(WeatherCode), weatherCode))
+        {
+            return ((WeatherCode)weatherCode).ToString().Replace('_', ' ');
+        }
+        else
+        {
+            return "Invalid weather code";
+        }
+    }
+    
+
+    public async Task<WeatherForecastApi?> GetWeatherForecastAsync(WeatherForecastOptionsApi optionsApi)
+    {
+        try
+        {
+            var response = await client.GetAsync(MergeUrlWithOptions(_weatherApiUrl, optionsApi));
+            response.EnsureSuccessStatusCode();
+
+            var weatherForecast = await JsonSerializer.DeserializeAsync<WeatherForecastApi>(
+                await response.Content.ReadAsStreamAsync(),
+                new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
```

Comment:
to chyba zbędne? Default dla web apps to case insensitive

Reply:
bez tego to niedziała masz coś konkretnego namyśli?

Reply:
no to jest dziwne, że bez tego nie działa, ale mozesz to olać i zostawić tak jak masz